### ISSUE TITLE
Mark flag mutually exclusive and required

### ISSUE THREAD

**Issue by shanduur-auto** [spf13/cobra#1761]
Hi!

I want to ask, if there is an option to mark one of `n` of the mutually exclusive flags required? This means there must appear exactly one of the `n` flags, otherwise an error should be printed.

For example:
```sh
# this should print contents from file:
$ echo -f ./file.txt
hello world

# this should act as a regular echo:
$ echo -i "hello world"
hello world

# this should print help and info about missing required flags:
$ echo
Error: one of mutually exclusive required flag(s) not set: "f", "i"
Usage:
  echo [flags]
```

**Comment by marckhouzam** [comment]
Hi @shanduur-auto.  Currently there is no special support from Cobra for what you are asking.  You would have to do the verification yourself in your program.

**Comment by marckhouzam** [comment]
Fixed by @marevers in #1952
